pkg/odo/cli/component/deploy.go

```diff
@@ -0,0 +1,195 @@
+	"github.com/openshift/odo/pkg/odo/util/completion"
+	"github.com/openshift/odo/pkg/odo/util/experimental"
+	"github.com/openshift/odo/pkg/util"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	odoutil "github.com/openshift/odo/pkg/odo/util"
+
+	ktemplates "k8s.io/kubectl/pkg/util/templates"
+)
+
+// TODO: add CLI Reference doc
+// TODO: add delete example
+var deployCmdExample = ktemplates.Examples(`  # Deploys an image and deploys the application 
+%[1]s
+  `)
+
+// DeployRecommendedCommandName is the recommended build command name
+const DeployRecommendedCommandName = "deploy"
+
+// DeployOptions encapsulates options that build command uses
+type DeployOptions struct {
+	*CommonPushOptions
+
+	// devfile path
+	DevfilePath     string
+	DockerfileURL   string
+	DockerfileBytes []byte
+	namespace       string
+	tag             string
+	ManifestSource  []byte
+}
+
+// NewDeployOptions returns new instance of BuildOptions
+// with "default" values for certain values, for example, show is "false"
+func NewDeployOptions() *DeployOptions {
+	return &DeployOptions{
+		CommonPushOptions: NewCommonPushOptions(),
+	}
+}
+
+// Complete completes push args
+func (do *DeployOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	do.DevfilePath = filepath.Join(do.componentContext, do.DevfilePath)
+	envInfo, err := envinfo.NewEnvSpecificInfo(do.componentContext)
+	if err != nil {
+		return errors.Wrap(err, "unable to retrieve configuration information")
+	}
+	do.EnvSpecificInfo = envInfo
+	do.Context = genericclioptions.NewDevfileContext(cmd)
+
+	return nil
+}
+
+// Validate validates the push parameters
+func (do *DeployOptions) Validate() (err error) {
+
+	// Validate the --tag
+	if do.tag == "" {
+		return errors.New("odo deploy requires a tag, in the format <registry>/namespace>/<image>")
+	}
+
+	err = util.ValidateTag(do.tag)
+	if err != nil {
+		return err
+	}
+
+	return
+}
+
+// Run has the logic to perform the required actions as part of command
+func (do *DeployOptions) Run() (err error) {
+	devObj, err := devfileParser.Parse(do.DevfilePath)
+	if err != nil {
+		return err
+	}
+
+	metadata := devObj.Data.GetMetadata()
+	dockerfileURL := metadata.Dockerfile
+	localDir, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	//Download Dockerfile to .odo, build, then delete from .odo dir
+	//If Dockerfile is present in the project already, use that for the build
+	//If Dockerfile is present in the project and field is in devfile, build the one already in the project and warn the user.
+	if dockerfileURL != "" && util.CheckPathExists(filepath.Join(localDir, "Dockerfile")) {
+		// TODO: make clearer more visible output
+		log.Warning("Dockerfile already exists in project directory and one is specified in Devfile.")
+		log.Warningf("Using Dockerfile specified in devfile from '%s'", dockerfileURL)
+	}
+
+	if !util.CheckPathExists(filepath.Join(localDir, ".odo")) {
+		return errors.Wrap(err, ".odo folder not found")
+	}
```

Comment:
This if statement is no longer required. It's remained from when we were downloading the Dockerfile onto disk

Reply:
Deleting as par of clean up PR